website/static/js/mailingListTreebeard.js

```diff
@@ -0,0 +1,163 @@
+'use strict';
+
+var $ = require('jquery');
+var m = require('mithril');
+var Treebeard = require('treebeard');
+var $osf = require('js/osfHelpers');
+var projectSettingsTreebeardBase = require('js/projectSettingsTreebeardBase');
+
+function expandOnLoad() {
+    var tb = this;  // jshint ignore: line
+    for (var i = 0; i < tb.treeData.children.length; i++) {
+        var parent = tb.treeData.children[i];
+        tb.updateFolder(null, parent);
+        expandChildren(tb, parent.children);
+    }
+}
+
+function expandChildren(tb, children) {
+    var openParent = false;
+    for (var i = 0; i < children.length; i++) {
+        var child = children[i];
+        var parent = children[i].parent();
+        if (child.data.kind === 'event' && child.data.event.notificationType !== 'adopt_parent') {
+            openParent = true;
+        }
+        if (child.children.length > 0) {
+            expandChildren(tb, child.children);
+        }
+    }
+    if (openParent) {
+        openAncestors(tb, children[0]);
+    }
+}
+
+function openAncestors (tb, item) {
+    var parent = item.parent();
+    if(parent && parent.id > 0) {
+        tb.updateFolder(null, parent);
+        openAncestors(tb, parent);
+    }
+}
```

Comment:
This file appears to be copied, pasted, and tweaked from `notificationsTreebeard.js`. Up to this point, the two files are identical. We should reuse the above three functions between the two files, rather than duplicating the functions. (It looks like `nodePrivacySettingsTreebeard.js` and `wikiSettingsTreebeard.js` also have some overlap; let's not worry about those here.) Options for deduplicating would be: a) functional reuse: moving the functions out into a third file (perhaps the oshHelpers junkdrawer, or a [new?] treebeard helpers file), b) object-oriented reuse: perhaps defining an `OsfTreebeard` object that extends the base `Treebeard`.